NStack/strings/ustring.cs

```diff
@@ -1224,6 +1221,16 @@ public int CompareTo (object value)
 			return 0;
 		}
 
+		int IComparable.CompareTo (object value)
+		{
+			if (value == null)
+				return 1;
+			var other = value as ustring;
+			if (other == null)
+				throw new ArgumentException ("Argument must be a ustring");
```

Comment:
Same here, the check needs to be:
```
if ((object)other  == null)
```

otherwise that triggers a call to operator ==.